### DISCUSSION TITLE
Test if package can be installed against current lockfile + vendor

### DISCUSSION THREAD

**Discussion by damienwebdev** [composer/composer#11141] [Q&A / Support] [answered]
I'm trying to write a fast "package installation" checker with composer and I'm wondering if there's a way to check whether a local package can be installed within another project "quickly".

Currently, in the dependent project I'm doing:

```bash
composer config repositories.local path path/to/package
composer require my-package "@dev"
```

Which works, but appears to trigger network requests (which I'm trying to avoid). Is something like this possible?

**Comment by Seldaek** [comment, marked as answer]
You could try to reduce the steps/narrow it down a bit more I guess:

```
composer config repositories.local path path/to/package
composer require my-package "@dev" --no-update
composer update my-package --dry-run
composer config repositories.local --unset
```

This will still do some network, but it should be less.

**Reply by damienwebdev** [reply]
Awesome, thanks @Seldaek
